pkg/reconciliation/construct_statefulset.go

```diff
@@ -27,7 +27,7 @@ func newNamespacedNameForStatefulSet(
 	dc *api.CassandraDatacenter,
 	rackName string) types.NamespacedName {
 
-	name := api.CleanupForKubernetes(dc.Spec.ClusterName) + "-" + dc.Name + "-" + rackName + "-sts"
+	name := api.CleanupForKubernetes(dc.Spec.ClusterName) + "-" + dc.Name + "-" + api.CleanupSubdomain(rackName) + "-sts"
```

Comment:
I think this could be simplified to:

```suggestion
	name := api.CleanupForKubernetes(dc.Spec.ClusterName + "-" + dc.Name + "-" + rackName + "-sts")
```

Because the real requirement is that `name` must conform to RFC 1035. IOW: if `dc.Spec.ClusterName` conforms to RFC 1035 and `rackName` conforms to RFC 1123, then the concatenation of both with a dash also conforms to RFC 1035.

Reply:
Actually, this resulted in incorrect result:

        	Error:      	Not equal:
        	            	expected: "cluster1-dc1-invalid-pod-name-sts"
        	            	actual  : "cluster1-dc1-invalid-pod-namests"

Reply:
I don't see why one dash would be removed and not the others, but OK.

Reply:
The 1035 does not allow dashes in every position (and not just the beginning). Not that I could directly say why that forth position from the end is not allowing it..